interpreter/luajit/luajit.go

```
@@ -0,0 +1,450 @@
+	h := fnv.New128a()
+	_, _ = h.Write([]byte(fileName)) //FIXME: needless allocation
+	_, _ = h.Write([]byte(funcName)) //FIXME: needless allocation
+	fileID, err := libpf.FileIDFromBytes(h.Sum(nil))
+	if err != nil {
+		return fmt.Errorf("failed to create a file ID: %v", err)
+	}
+	trace.AppendFrame(libpf.LuaJITFrame, fileID, libpf.AddressOrLineno(pc))
+	symbolReporter.FrameMetadata(fileID, libpf.AddressOrLineno(pc), util.SourceLineno(line),
+		pc, funcName, fileName)
+	return nil
+}
+
+// Symbolize is a little weird in lua since we need the caller frame to get the name of the
+// function being called. So we stash the info for the current frame and each symbolize call
+// actually symolizes the previous frame.  The unwinder emits a special frame with file 0 to
+// indicate the end of the lua stack.
+func (l *luajitInstance) Symbolize(symbolReporter reporter.SymbolReporter, frame *host.Frame,
+	trace *libpf.Trace) error {
+	if !frame.Type.IsInterpType(libpf.LuaJIT) {
+		return interpreter.ErrMismatchInterpreterType
+	}
+
+	// We need the calling frame to find the name of the current frame so save it and do it on next
+	// frame. This assumes that the harness calls Symbolize from top to bottom of the stack in
+	// order!!  The final luajit frame will have a 0 frame.File.
+	if l.previousFrame == nil {
+		l.previousFrame = frame
+		logf("lj: new LuaJIT frame")
+		return nil
+	}
+
+	// The function being invoked at this frame (caller) is the name for the previous
+	// frame (callee). For the last frame frame.File will be 0 and pt will be nil and funcName
+	// will be "main".  Lua is a real deal dynamic functional language, functions don't have
+	// inherent names they are just values.
+	var funcName string
+	if frame.File == C.LUAJIT_FFI_FUNC {
+		switch frame.Lineno & 7 {
+		case 1:
+			funcName = "c-frame"
+		case 2:
+			funcName = "cont-frame"
+		case 3:
+			panic("unexpected frame type 3")
```

Why panic instead of return an error?

---

Kind of a debugging scapel, it should never happen.  I'll replace panic with an error return.